src/main/java/net/yogstation/yogbot/listeners/commands/ManageEchelonCommand.kt

```diff
@@ -0,0 +1,135 @@
+			INSERT INTO ${databaseManager.prefix("bound_credentials")} (`ckey`, `flags`) VALUES (?, 'allow_proxies')
+		""".trimIndent(), Statement.RETURN_GENERATED_KEYS).use { preparedStatement ->
+			preparedStatement.setString(1, StringUtils.ckeyIze(args[2]))
+			try {
+				val rows = preparedStatement.executeUpdate()
+				if(rows == 0) throw SQLException("No rows inserted")
+			} catch (e: SQLException) {
+				return DiscordUtil.reply(event, "An error has occurred: ${e.message}")
+			}
+			preparedStatement.generatedKeys.use {
+				if(!it.next()) return DiscordUtil.reply(event, "No ID returned")
+				return DiscordUtil.reply(event, "Created exception ${it.getLong(1)} for `${args[2]}`.")
+			}
+		}
+		}
+	}
+
+	private fun searchWhitelist(event: MessageCreateEvent, args: Array<String>): Mono<*> {
+		if(args.size < 3) return DiscordUtil.reply(event, "Usage is `${args[0]} ${args[1]} <ckey>`")
+		databaseManager.byondDbConnection.use {connection -> connection.prepareStatement(
+			"""
+				SELECT `id`, `ckey`
+				FROM ${databaseManager.prefix("bound_credentials")}
+				WHERE ckey = ? AND FIND_IN_SET('allow_proxies', flags)
+			""".trimIndent()
+		).use { preparedStatement ->
+			preparedStatement.setString(1, args[2])
+			preparedStatement.executeQuery().use { results ->
+				val builder = StringBuilder("Whitelists for ckey `${args[2]}`:\n```\n")
+				val monos: MutableList<Mono<*>> = ArrayList();
+				var first = true
+				while(results.next()) {
+					builder.append("${results.getInt("id")}: Ckey ${results.getString("ckey")}\n")
+					if(builder.length > Yogbot.MAX_MESSAGE_LENGTH - 10) {
+						builder.append("```")
+						monos.add(if(first) DiscordUtil.reply(event, builder.toString()) else DiscordUtil.send(event, builder.toString()))
+						first = false
+						builder.setLength(0)
+						builder.append("```")
+					}
+				}
+				builder.append("```")
+				monos.add(if(first) DiscordUtil.reply(event, builder.toString()) else DiscordUtil.send(event, builder.toString()))
+				return Mono.`when`(monos)
+			}
+		}
+		}
+	}
+
+	private fun removeWhitelist(event: MessageCreateEvent, args: Array<String>): Mono<*> {
+		if(args.size < 3) return DiscordUtil.reply(event, "Usage is `${args[0]} ${args[1]} <whitelist>`")
+		val id = args[2].toIntOrNull() ?: return DiscordUtil.reply(event, "${args[2]} is not a valid integer")
+		val monos: MutableList<Mono<*>> = ArrayList()
+		databaseManager.byondDbConnection.use {  connection ->
+			var oldFlags = ""
+			connection.prepareStatement("""
+				SELECT `ckey`, `flags`
+				FROM ${databaseManager.prefix("bound_credentials")}
+				WHERE `id` = ? AND FIND_IN_SET('allow_proxies', flags)
+			""".trimIndent()).use { preparedStatement ->
+				preparedStatement.setInt(1, id)
+				preparedStatement.executeQuery().use {
+					if(!it.next()) return DiscordUtil.reply(event, "No binding with id $id found!")
+					monos.add(DiscordUtil.reply(event, "Deleted whitelist for ckey `${it.getString("ckey")}`"))
```

Comment:
Does this work properly when therse an error in later lines? I dont know what mono does